web/db/actions/Post.ts

```diff
@@ -82,6 +90,54 @@ async function getResizedUploadUrl(uuid: string): Promise<string> {
   return `${consts.baseUrl}/api/resizedUpload/${token}`;
 }
 
+async function deleteAttachments(keysToDelete: string[]) {
+  let numTries = 1;
+  const MAXTRIES = 3;
+  const objectsToDelete = keysToDelete.map((keyToDelete) => ({
+    Key: keyToDelete,
+  }));
+  const deleteObjectsCommand = new DeleteObjectsCommand({
+    Bucket: consts.storageBucket,
+    Delete: { Objects: objectsToDelete, Quiet: true },
+  });
+  while (numTries <= MAXTRIES) {
+    try {
+      const returned = await storageClient.send(deleteObjectsCommand);
+      if (!returned["Deleted"]) {
+        return { success: true };
+      }
+    } catch (error) {
+      //TODO: Write cron job to mark unsuccessful deletions to delete later
+      throw new Error("All attachments not successfully deleted");
+    }
+    numTries++;
+  }
+  return { success: true };
+}
+
+async function deletePost(id: ObjectId) {
+  try {
+    const post = await getPost(id, false);
+    const returned = await deleteAttachments(post.attachments);
+    if (returned.success) {
```

Comment:
You can remove this since we either throw an error or it succeeds. If it errors, we enter the catch anyway.